src/goto-checker/incremental_goto_checker.h

```diff
@@ -83,6 +87,8 @@ class incremental_goto_checkert
   const optionst &options;
   ui_message_handlert &ui_message_handler;
   messaget log;
+
+  std::shared_ptr<solver_hardnesst> solver_hardness;
```

Comment:
I think you explained this at some point but I still don't quite get why this is shared rather than unique. Do all the things that want a handle on this thing have different/non-hierarchical lifetimes?